lib/queue.js

```diff
@@ -1,21 +1,26 @@
+'use strict'
+
 const spq = require('shuffled-priority-queue')
+const lru = require('hashlru')
 const { EventEmitter } = require('events')
 const peerInfo = require('./peer-info')
 const timer = require('./bulk-timer')
 
 module.exports = (opts) => new PeerQueue(opts)
 
 class PeerQueue extends EventEmitter {
-  constructor (opts) {
+  constructor (opts = {}) {
     super()
 
-    if (!opts) opts = {}
+    const {
+      requeue = [ 1000, 5000, 15000 ],
+      cacheSize = 100
+    } = opts
 
     this.destroyed = false
-    this._infos = new Map() // TODO: turn into LRU
+    this._infos = lru(cacheSize)
```

Comment:
i don't this this should be an LRU actually. The queue has a .remove method for removal. No implicit removal should be done.

Reply:
ok, scenarios for explicit removal

* when a peer has been banned?
* when the cache exceeds a memory limit or peer count?
* any others?